server.js

```diff
@@ -55,17 +51,8 @@ const server = http
     return app(req, res);
   })
   .on("listening", () => {
-    console.log("  -------------------------------------------------------------------------");
-    console.log(nextstrainAbout);
-    console.log(`  Server listening on port ${server.address().port}`);
-    console.log(`  Accessible at https://nextstrain.org or http://localhost:${server.address().port}`);
-    console.log(`  Server is running in ${app.locals.production ? 'production' : 'development'} mode`);
-    if (app.locals.STATIC_SITE_PRODUCTION) {
-      console.log(`  Next.js frontend is running in production mode (statically generated assets)`)
-    } else {
-      console.log(`  Next.js frontend is running in development mode (hot reloading etc)`)
-    }
-    console.log("\n  -------------------------------------------------------------------------\n\n");
+    utils.log(`Server now listening on port ${server.address().port} (startup message will appear shortly)`);
+    startupMessage(server, app, args.app!==DEFAULT_ENTRYPOINT)
```

Comment:
_(non-blocking)_

The mapping of `args.app!==DEFAULT_ENTRYPOINT` to `groupsApp` (from the function signature) feels indirect and implicit based on knowledge outside this file that there is only one other possible entrypoint which is for the groups app. Why not change the option from `--app` to `--groupsApp` as done in c125524212d5091802aa6888f691edf33aef7590?

Reply:
Because that's a breaking change for external users. But you're involved in these use cases (I'm not!), so I'm happy to take your advice and make this change here.

Reply:
People running the groups app should be following [groupsApp.md](https://github.com/nextstrain/nextstrain.org/blob/2b12ed80595a1bdba7a4e8c4cb045cb84470f618/groupsApp.md?plain=1) which uses `npm run groups`. As long as we change the command used by `npm run groups`, it shouldn't be a breaking change.

Reply:
I'd consider this blocking. `args.app!==DEFAULT_ENTRYPOINT` mapped to "Groups app or not" is sloppy. How about reporting the entrypoint instead? And if you really want to do keep wording like `NEXTSTRAIN ' + (groupsApp?'GROUPS ':'') + 'SERVER'`, then have a property on `app` itself (i.e. set by each app constructor) with the desired title, e.g.

```diff
diff --git a/src/app.js b/src/app.js
index dcb8844a..634ef8e3 100644
--- a/src/app.js
+++ b/src/app.js
@@ -29,7 +29,7 @@ const {
  *   /logged-in
  *   /logout
  */
-const app = setupApp();
+const app = setupApp("Nextstrain Server");
 
 
 /* Canary.
diff --git a/src/groupsApp.js b/src/groupsApp.js
index 0430c729..fb5275a3 100644
--- a/src/groupsApp.js
+++ b/src/groupsApp.js
@@ -21,7 +21,7 @@ const {
  *   /logged-in
  *   /logout
  */
-const app = setupApp();
+const app = setupApp("Nextstrain Groups Server");
 
 
 /* Charon API used by Auspice.
diff --git a/src/routing/setup.js b/src/routing/setup.js
index 95f024f7..70eef7b7 100644
--- a/src/routing/setup.js
+++ b/src/routing/setup.js
@@ -11,7 +11,7 @@ import { replacer as jsonReplacer } from '../json.js';
 import * as middleware from '../middleware.js';
 
 
-export function setupApp() {
+export function setupApp(name) {
 
   /* Express boilerplate.
    */
@@ -19,6 +19,7 @@ export function setupApp() {
 
   app.set("json replacer", jsonReplacer);
 
+  app.locals.name = name;
   app.locals.production = PRODUCTION;
   app.locals.STATIC_SITE_PRODUCTION = STATIC_SITE_PRODUCTION;
 
diff --git a/src/startupMessage.js b/src/startupMessage.js
index b7334a1e..11b14018 100644
--- a/src/startupMessage.js
+++ b/src/startupMessage.js
@@ -7,9 +7,9 @@ export const nextstrainAbout = `
   See https://github.com/nextstrain/nextstrain.org for more.
 `;
 
-export async function startupMessage(server, app, groupsApp) {
+export async function startupMessage(server, app) {
 
   console.log("  -------------------------------------------------------------------------");
-  console.log('                         NEXTSTRAIN ' + (groupsApp?'GROUPS ':'') + 'SERVER');
+  console.log('                         ' + app.name.toUpperCase());
 
   console.log(nextstrainAbout);
   console.log(`  Server listening on port ${server.address().port}, accessible at http://localhost:${server.address().port}`);
```

I'd keep `--app`, i.e. not switch it to `--groupsApp`. I'm not inclined to break the invocation for others for what feels like just churn.

Reply:
@victorlin 

> People running the groups app should be following [groupsApp.md](https://github.com/nextstrain/nextstrain.org/blob/2b12ed80595a1bdba7a4e8c4cb045cb84470f618/groupsApp.md?plain=1) which uses `npm run groups`. As long as we change the command used by `npm run groups`, it shouldn't be a breaking change.

I dunno, that seems short-sighted to me. That document is a guide/how-to, not a mandate to run things exactly a certain way…

Reply:
That's a fair point. I don't like the idea of setting a new property `app.name` for it to only be used in the startup message. Reporting the entrypoint file seems the most direct with minimal changes.

Reply:
I've gone back to my original implementation. We now attempt to report the AWS credentials for all use cases, thus avoiding the conditional logic being debated here.